server/modules/authentication/eveonline/authentication.js

```diff
@@ -0,0 +1,60 @@
+const _ = require('lodash')
+/* global WIKI */
+
+// ------------------------------------
+// EveOnline Account
+// ------------------------------------
+
+const OAuth2Strategy = require('passport-oauth2').Strategy
+var jwt = require('jsonwebtoken');
+
+
+module.exports = {
+  init (passport, conf) {
+    OAuth2Strategy.prototype.userProfile = function (accessToken, cb) {
+      try {
+        const payload = jwt.decode(accessToken);
+        var sub=payload.sub;
+        var chardetail=sub.split(":");
+        var charid=chardetail[2];
+        cb(null, {
+          id: charid,
+          displayName: payload.name,
+          email: charid+'@auth.eveonline.com',
+        })
+      } catch (err) {
+        WIKI.logger.warn('Eve Online - Failed to parse user profile.')
```

Comment:
Using this in version 2.5.297, I needed to comment out this logger line to get the module to function.